app/src/test/java/com/android/streetworkapp/model/park/ParkViewModelTest.kt

```diff
@@ -187,4 +191,73 @@ class ParkViewModelTest {
     testDispatcher.scheduler.advanceUntilIdle()
     verify(repository).deleteParkByPid(pid)
   }
+
+  @Test
+  fun setCurrentParkUpdatesCurrentPark() {
+    val park =
+        Park(
+            pid = "123",
+            name = "Sample Park",
+            location = ParkLocation(0.0, 0.0, "321"),
+            imageReference = "parks/sample.png",
+            rating = 4.0f,
+            nbrRating = 2,
+            capacity = 10,
+            occupancy = 5,
+            events = listOf("event1", "event2"))
```

Comment:
Suggestion: Since SonarCloud will likely detect these lines as duplicates, you could create a` private fun` to handle the park creation.

I would do something like this:
```kotlin
private fun createPark(
    pid: String = "123",
    locationId: String = "321"
) = Park(
    pid = pid,
    name = "Sample Park",
    location = ParkLocation(0.0, 0.0, locationId),
    imageReference = "parks/sample.png",
    rating = 4.0f,
    nbrRating = 2,
    capacity = 10,
    occupancy = 5,
    events = listOf("event1", "event2")
)
```

Reply:
Good suggestion! I've made the change.

Reply:
Perfect thread resolved !